### DISCUSSION TITLE
Issues regarding renaming of table name

### DISCUSSION THREAD

**Discussion by ankitkpandey1** [sqlalchemy/alembic#1132] [Usage Questions]
Hi,

There is a use case where I need to rename a table name. Since alembic doesn't support detection of renamed tables (via auto generate), I custom wrote a migration script that calls  `op.rename_table`. This worked fine for the running system which had table named initially as X and after migration, got changed to Y.

There is a problem if I want to use this script for a fresh system (which doesn't have any notion of Table X as it is renamed now), it fails as it tries to access non existing table X. 

I would be glad if someone can suggest a better approach or any alteration to the current one.

Thanks

**Comment by CaselIT** [comment]
Hi
> 
> There is a problem if I want to use this script for a fresh system (which doesn't have any notion of Table X as it is renamed now), it fails as it tries to access non existing table X.

This is not limited to rename table, pretty much every migration operation is not replayable on a database that's already updated for similar reasons

If you plan on creating a database from scratch using `MetaData.create_all` then you will need to stamp it with the current alembic version, so that alembic knows that it should not try to apply old migrations, since they where already applied so to speak.

You can use the [stamp command](https://alembic.sqlalchemy.org/en/latest/api/commands.html#alembic.command.stamp) for it
